timApp/static/scripts/tim/main.ts

```diff
@@ -148,7 +151,9 @@ function createDowngradedAppModule() {
     doDowngrade(dg, "timSearchButton", SearchButtonComponent);
     doDowngrade(dg, "timRoleInfo", RoleInfoComponent);
     doDowngrade(dg, "csFormulaEditorLoader", FormulaEditorLoaderComponent);
-
+    doDowngrade(dg, "timUserProfile", UserProfileComponent);
+    doDowngrade(dg, "timParticipantList", ParticipantListComponent);
+    doDowngrade(dg, "timCourseManager", CourseManagerComponent);
```

Comment:
Ei tarvitse tähän PRään tehdä, mutta kannattaa kirjata ylös ja korjata jossain vaiheessa: `doDowngrade`:n kutsuminen tässä liittää komponentin koodin pääbundleen (eli siihen JS-tiedostoon, jonka selain lataa aina, kun TIM avataan).  Pääbundle kannattaa pitä pienenä, se on jo muutenkin vähän liian raskas.

Jos näitä komponentteja ei tarvitse joka ikisellä sivulla, niin eräs tapa toimia on seuraava:

1. Tee erillinen `NgModule`, jossa on haluamasi komponentit ja kutsu niille erikseen `doDowngrade`. Katso esim. `TimArchiveModule` <https://github.com/TIM-JYU/TIM/blob/d62cd7e664b400f35d719989e72781470fbfd7b1/timApp/static/scripts/tim/messaging/archive/tim-archive.module.ts#L13-L33>
2. Lisää dynaaminen import `staticDynamicModules.ts`ään. Katso esim. edellinen `TimArchiveModule`n tapauksessa: <https://github.com/TIM-JYU/TIM/blob/970793c945fe62cd81dcfc649748604b3ace07c4/timApp/static/scripts/tim/staticDynamicImport.ts#L83-L84>
3. Niissä dokumenteissa, jossa sitten tarvitaan moduulissa olevat komponentit, lisätään dokumentin asetuksiin tai preambleen

    ```yaml
    additional_angular_modules:
      - timArchive
    ```
    jolloin TIM lataa moduulin ja sen sisältämät komponentit vain kyseisessä dokumenteissa.

"Paras" vaihtoehto on toki tehdä näistäkin komponenteista plugineja, jolloin niitä ladataan aina dynaamisesti sen mukaan, käytetäänkö pluginia dokumentissa. 

Tätä ei tarvitse toteuttaa tässä PRssä, mutta voisi kirjata johonkin TODOksi ja kuitata tämä kommentti sillä.